inference-engine/src/mkldnn_plugin/nodes/mkldnn_if_node.cpp

```diff
@@ -165,57 +162,86 @@ void MKLDNNIfNode::initSupportedPrimitiveDescriptors() {
     supportedPrimitiveDescriptors.emplace_back(config, impl_desc_type::unknown);
 }
 
-
 void MKLDNNIfNode::createPrimitive() {
-    const auto& eng = getEngine();
+    if (inputShapesDefined()) {
+        if (needPrepareParams())
+            prepareParams();
+        updateLastInputDims();
+    }
+}
 
-    for (auto& map_rule : thenInputPortMap) {
-        auto &fromMem = getParentEdgesAtPort(map_rule.from)[0]->getMemoryPtr();
-        auto &toMem = inputMemThen[map_rule.to];
+bool MKLDNNIfNode::needPrepareParams() const {
+    return MKLDNNNode::needPrepareParams() || condition != (reinterpret_cast<const bool*>(getParentEdgeAt(0)->getMemoryPtr()->GetPtr()))[0];
```

Comment:
better to call `inputShapeModifed` insted `MKLDNNNode::needPrepareParams`

Reply:
Sure, thanks